data_storage/export_format.md

```diff
@@ -1,22 +1,28 @@
 Export Format
 -------------
 
-- **5.1.1** ![](/badge/req.png) A Toxcore data file saved in the root of the client's
-  working directory, with the file name `<profile_name>.tox`
+- **5.1.1** ![](/badge/req.png) The client will export files relative to a folder called
+  the export folder, which must be in a location accessible to the user.
 
-- **5.1.2** ![](/badge/req.png) If the client supports user avatars, they will be stored
-  under the `avatars/` directory in the root of the client's working directory.
+- **5.1.2** ![](/badge/rec.png) The user should be able to choose the location
+  of the export folder.
 
-- **5.1.3** ![](/badge/req.png) Avatars are named according to the following format:
+- **5.1.3** ![](/badge/req.png) A Toxcore data file saved in the root of the
+  export folder, with the file name `<profile_name>.tox`.
```

Comment:
>I think we disagree on what is toxcore's job. I'm on the far end of "should provide networking facilities only". Ideally, it does exactly that and specialises on it, while other libraries built on top can provide default implementations of standardised behaviour (i.e. things in TCS and related documents).

That's a reasonable stance, but I think we might disagree on the details. 

I'm happy with complementary standardized libraries building on top of toxcore to providing implementations of standard behavior that all clients may reasonably need, in that I consider things like libtoxencryptsave or libtoxdns helpful and part of "Toxcore" even though it's not strictly in libtoxcore. 
I would be also fine with new standardized libraries providing a reference implementation of new features, even if they don't have "tox" or "toxcore" in their name.

On the other hand, I'm very wary of trying to standardize third-party libraries that I think clearly overlap with the libtoxcore API. Having competing implementations of the same basic features in different libraries, with completely different file formats and even different supported feature sets (which might diverge further), is not a situation I at all want to be in. That way lies madness and weak standards.

>Indeed not everything that's in the tox save data can be extracted, but it's possible to get a tox instance to its original state (modulo connected DHT nodes) using API calls and not using save data.

That's true, and arguably new APIs could be added to toxcore to extract the missing data, but that's not what I strongly disagree on.

>I would like to see the save data going away or using a format I can easily inspect and modify from code without manually writing specialised parsers and generators for it.

I would ask, what is preventing you from using the Toxcore API or one of its binding?
*Why would you not rather fix those problems in toxcore?*
Is it really so bad that you would rather start an alternative library than work with it? 

If there is any problem with Toxcore, *this is something I would take very seriously* and work towards fixing, because those sort of problems do not get better with time.
If there is nothing wrong with Toxcore, then I'm even more interested in your answers.

>However, that discussion doesn't belong here, so given the current state of toxcore, we might as well make use of what's there.

I'm always happy to talk if there is an issue!
If we can't reach a consensus, that won't be a very good standard, will it?

>The only thing that worries me is that if we agree to use the blob created by toxcore, any change to that decision will create a backwards-incompatible change in the standard.

Then this is exactly the time to fix it.
Whether we standardize on a third party library, or on fixing the toxcore blob format to become backwards-compatible, in both cases older clients will lose forward-compatibility with new saves.
If we're going to break things, after the standard is published will be too late.